fractal_tasks_core/lib_zarr.py

```diff
@@ -0,0 +1,245 @@
+        logger.error(error_msg)
+        raise OverwriteNotAllowedError(error_msg)
+
+
+def _write_elem_with_overwrite(
+    group: zarr.Group,
+    key: str,
+    elem: Any,
+    *,
+    overwrite: bool,
+    logger: Optional[logging.Logger] = None,
+) -> None:
+    """
+    Wrap `anndata.experimental.write_elem`, to include `overwrite` parameter.
+
+    See docs for the original function
+    [here](https://anndata.readthedocs.io/en/stable/generated/anndata.experimental.write_elem.html).
+
+    This function writes `elem` to the sub-group `key` of `group`. The
+    `overwrite`-related expected behavior is:
+
+    * if the sub-group does not exist, create it (independently on
+      `overwrite`);
+    * if the sub-group already exists and `overwrite=True`, overwrite the
+      sub-group;
+    * if the sub-group already exists and `overwrite=False`, fail.
+
+    Note that this version of the wrapper does not include the original
+    `dataset_kwargs` parameter.
+
+    Args:
+        group:
+            The group to write to.
+        key:
+            The key to write to in the group. Note that absolute paths will be
+            written from the root.
+        elem:
+            The element to write. Typically an in-memory object, e.g. an
+            AnnData, pandas dataframe, scipy sparse matrix, etc.
+
+    Raises:
+        OverwriteNotAllowedError:
+            If `overwrite=False` and the sub-group already exists.
+    """
+
+    # Set logger
+    if logger is None:
+        logger = logging.getLogger(None)
+
+    if key in set(group.group_keys()):
+        if not overwrite:
+            error_msg = (
+                f"Sub-group '{key}' of group {group.store.path} "
+                f"already exists, but `{overwrite=}`. "
+                "Hint: try setting `overwrite=True`."
+            )
+            logger.error(error_msg)
+            raise OverwriteNotAllowedError(error_msg)
+    write_elem(group, key, elem)
+
+
+def write_table(
+    image_group: zarr.Group,
+    table_name: str,
+    table: ad.AnnData,
+    overwrite: bool = False,
+    ngff_roi_attrs: Optional[dict[str, str]] = None,
+    logger: Optional[logging.Logger] = None,
+) -> zarr.group:
+    """
+    FIXME
+
+    Args:
+        image_group:
+            TBD
+        table_name:
+            TBD
+        table:
+            TBD
+        overwrite:
+            TBD
+        ngff_roi_attrs:
+            TBD
+        logger:
+            TBD
+
+    Returns:
+        Zarr group of the new table.
+    """
+
+    # Set logger
+    if logger is None:
+        logger = logging.getLogger(None)
+
+    # Create tables group, if needed
+    if "tables" not in set(image_group.group_keys()):
+        tables_group = image_group.create_group("tables", overwrite=False)
+    else:
+        tables_group = image_group["tables"]
+
+    # Check whether subgroup already exists, and proceed accordingly
+    if table_name in set(tables_group.group_keys()) and not overwrite:
+        raise
```

Comment:
What does a simple raise without an error message do? Should we specify here that a table couldn't be overwritten?

Reply:
Of course. This was just the structure of the function, to correctly set its scope. The actual error will be an appropriate one.